wordpress-mobile/WordPress-iOS
# Prompts: Show Bloganuary modal when Learn more is tapped

Addresses #22127
Depends on #22121

This PR adds the Bloganuary modal screen, displays a slightly different variation of text depending on the prompt dashboard card state, and implements the tap action when the primary button is tapped. When the prompt card is disabled, tapping on the button should shortly enable the prompt card on the dashboard.

Note that tracking is excluded from this PR. I'll address them there along with any accessibility issues for both the card and the modal component.

> [!NOTE]
> I added a new local feature flag, `bloganuaryCardDebugOverride`, which aims to simplify testing without changing anything in the code. When this is turned on, the card should always show (except when it's set to hidden from the ellipsis menu). This flag is enabled by default for local and branch test builds, and I'll remove them on the PR that enables the remote feature flag.

## To test

- Launch the Jetpack app.
- Log in with your WP.com account.
- Turn off the prompts dashboard card from the ellipsis menu.
- Tap 'Learn more' on the Bloganuary dashboard card.
- 🔎  Verify that the modal screen is shown.
- 🔎  Verify that the primary button says 'Turn on blogging prompts'. 
- Tap the primary button.
- 🔎 Verify that the modal is dismissed, and the prompts dashboard card now appears.
- Tap 'Learn more' on the Bloganuary dashboard card again.
- 🔎  Verify that the primary button says `Let's go!`.
- Tap the primary button. 
- 🔎 Verify that the modal is dismissed

Please test the modal in portrait / landscape orientations, and in iPad devices.

## Regression Notes
1. Potential unintended areas of impact
Should be none. The modal screen is completely new and doesn't touch any of the existing code.

2. What I did to test those areas of impact (or what existing automated tests I relied on)
Manually tested the changes.

3. What automated tests I added (or what prevented me from doing so)
N/A.

PR submission checklist:


UI Changes testing checklist:
